### PR TITLE
Update README.md to explain kubernetes tags and the v5.0.0 tag

### PR DESCRIPTION
**Pull request by caesarxuchao**
@cheftako could you help proof read? Thank you.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,27 +81,26 @@ We will backport bugfixes--but not new features--into older versions of
...
+* `+` client-go has features or API objects that may not be present in the
   Kubernetes cluster, but everything they have in common will work. Please
   note that alpha APIs may vanish or change significantly in a single release.
-* `†` client-go has new features or api objects, and some APIs running in the
-  cluster may have been deprecated and removed from client-go. But everything
-  they share in common (i.e., most APIs) will work.
 * `-` The Kubernetes cluster has features the client-go library can't use
-  (additional API objects, etc).
+  (additional new API, or deprecated API, etc). But everything
+  they share in common (i.e., most APIs) will work.
```
Comment (cheftako) [inline README.md]: I suspect we are missing the case where an older version of the client has an API which is no longer supported by the the newer cluster. This is similar to the '+' case (new client with a cluster so old it does not yet support the feature). However I do not see any such warning for things line client 1.4 with Kubernetes 1.8, so I suspect we are missing this case.

Comment (caesarxuchao) [reply README.md]: Comments addressed. PTAL. Thank you.

File: README.md [hunk 2]
```diff
@@ -80,27 +81,29 @@ We will backport bugfixes--but not new features--into older versions of
...
-  Kubernetes cluster, but everything they have in common will work. Please
-  note that alpha APIs may vanish or change significantly in a single release.
-* `†` client-go has new features or api objects, and some APIs running in the
-  cluster may have been deprecated and removed from client-go. But everything
-  they share in common (i.e., most APIs) will work.
-* `-` The Kubernetes cluster has features the client-go library can't use
-  (additional API objects, etc).
+* `+` client-go has features or API objects that may not be present in the
+  Kubernetes cluster, either due to that client-go has additional new API, or
+  that the server has deprecated old API. However, everything they have in
```
Comment (cheftako) [inline README.md]: More than deprecated it must have actually been removed.

File: README.md [hunk 3]
```diff
@@ -80,27 +81,29 @@ We will backport bugfixes--but not new features--into older versions of
...
+  common (i.e., most APIs) will work. Please note that alpha APIs may vanish or
+  change significantly in a single release.
+* `-` The Kubernetes cluster has features the client-go library can't use,
+  either due to the server has additional new API, or that client-go has
+  deprecated old API. However, everything they share in common (i.e., most APIs)
```
Comment (cheftako) [inline README.md]: Again more than just deprecated

File: README.md (hunk 2 above)
Comment (caesarxuchao) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (caesarxuchao) [reply README.md]: Done.

**Comment by caesarxuchao** [comment]
Thanks @cheftako for the review. Merging.

**Pull request merged**